Library/Homebrew/cmd/update.sh

```diff
@@ -564,7 +568,26 @@ EOS
   rm -f "${update_failed_file}"
   rm -f "${missing_remote_ref_dirs_file}"
 
-  for DIR in "${HOMEBREW_REPOSITORY}" "${HOMEBREW_LIBRARY}"/Taps/*/*
+  DIRS=("${HOMEBREW_REPOSITORY}")
+
+  # If the user specified any taps by name with -t or --tap,
+  # include only those taps in our list of DIRS to update.
+  if [[ "${#ALLOWLISTED_TAPS[@]}" == "0" ]]
+  then
+    DIRS+=("${HOMEBREW_LIBRARY}"/Taps/*/*)
+  else
+    for DIR in "${HOMEBREW_LIBRARY}"/Taps/*/*
+    do
+      local relative_dir="${DIR#"${HOMEBREW_LIBRARY}/Taps/"}"
+      local tap="${relative_dir/homebrew-/}"
+      if [[ " ${ALLOWLISTED_TAPS[*]} " == *" ${tap} "* ]]
+      then
+        DIRS+=("${DIR}")
+      fi
```

Comment:
NOTE: This implementation ignores invalid taps. I think this is correct b/c it's treating `"${HOMEBREW_LIBRARY}"/Taps/*/*` as the source-of-truth and the users' arguments as a filter, but do you think it should print a warning in this scenario? Like:
```
$ brew update --tap=homebrew/core --tap=nope/nope
Tap nope/nope is not installed.
You can tap it with brew tap nope/nope.
```